web-client/src/components/SignInForm.jsx

```diff
@@ -0,0 +1,215 @@
+            onSubmit={handleSubmit((data) => {
+              if (AreEmailAndPasswordCorrect()) {
+                SetMessageStyle(correctStyle);
+                alert(JSON.stringify(data));
+              } else SetMessageStyle(incorrectStyle);
+            })}
+          >
+            {/* Login/Email Input */}
+            <TextField
+              className={style.textField}
+              inputRef={register}
+              required
+              id="email-signin"
+              name="email"
+              label="Login/Email"
+              type="email"
+              value={values.email}
+              autoComplete="email"
+              onChange={handleChange("email")}
+              InputLabelProps={{
+                shrink: true,
+              }}
+              variant="outlined"
+              error={!!errors.email}
+              helperText={errors?.email?.message}
+            />
+            {/* Password Input */}
+            <TextField
+              className={style.textField}
+              inputRef={register}
+              required
+              id="password-signin"
+              name="password"
+              label="Password"
+              type="password"
+              value={values.password}
+              autoComplete="current-password"
+              onChange={handleChange("password")}
+              InputLabelProps={{
+                shrink: true,
+              }}
+              variant="outlined"
+              error={!!errors.password}
+              helperText={errors?.password?.message}
+            />
+            {/* Info about correct password and email */}
+            <FormHelperText
+              error={true}
+              style={_messageStyle}
+              className={style.formHelperText}
+            >
+              {"Error: Incorrect password or/and email"}
+            </FormHelperText>
+            <div>
+              {/* Forgot password */}
+              <PopUpForgotPassword SET_EMAIL={passEmail} />
+              {console.log(_forgotEmail)}
+              {/* Button Submit */}
+              <Button
+                className={style.btnSignIn}
+                color="primary"
+                type="submit"
+                variant="contained"
+              >
+                Sign in
+              </Button>
+            </div>
+          </form>
+        </div>
+
+        {/* Info about signing up */}
+        <div className={style.signUp}>
+          <h2 className={style.heading}>{props.heading}</h2>
+          <p className={style.content}>{props.content}</p>
+          <Link to={props.signUpLink} style={{ textDecoration: "none" }}>
+            <Button
+              variant="outlined"
+              color="primary"
+              className={style.btnSignUp}
+            >
+              {props.btn}
+            </Button>
+          </Link>
+        </div>
+      </div>
+      <span className={style.bottomMessage}>
+        If you want to sign in as {props.signInAs}, Sign in{" "}
```

Comment:
`{" "}` ?

Reply:
This is a space at the end of the line. It was added by autoformat.
Instead, I would have to write:
`<span className={style.bottomMessage}>If you want to sign in as {props.signInAs}, Sign in </span>`
in one line.

Reply:
It looks funny, but okay :p 
https://stackoverflow.com/questions/40264084/best-practice-when-adding-whitespace-in-jsx